packager/media/formats/mp2t/ts_segmenter.cc

```diff
@@ -181,6 +183,15 @@ Status TsSegmenter::FinalizeSegment(uint64_t start_timestamp,
   // This method may be called from Finalize() so ts_writer_file_opened_ could
   // be false.
   if (ts_writer_file_opened_) {
+    // RENAME FILE:
+    std::string segment_name_segment_index =
+        GetSegmentName(muxer_options_.segment_template, next_pts_,
+                       segment_index - 1, muxer_options_.bandwidth);
+
+    if (!ts_writer_->RenameFile(segment_name_segment_index)) {
+      LOG(ERROR) << "TS Error renaming the file";
+    }
```

Comment:
Unfortunately, renaming the file after the segment is completed won't work for "Low Latency Live Streaming" we want to achieve later.

We have two options: 

(1) Store the segment data in a buffer first, and write it to the file only after the segment is completed.
     - We could either store the data in a list if MediaSamples or 
     - Write the samples to a uint8_t byte buffer.
 (2) Add a new kSegmentInfoStart stream data type and send it when the segment begins in ChunkingHandler

Reply:
For option (1), how about using BufferWriter as a member in Writer classes to store the bytes to be written in file and then in FinalizeSegment, this buffer can be written to the file? Does that make sense for both mpegts (write to buffer instead of file in WritePesToFile) and webm (MkvWriter::Write and MkvWriter::WriteFromFile)?

Reply:
Have updated the PR by reverting the file renaming and used buffer for mp2ts and webm.